### PR TITLE
Register resource classes as CDI beans when not annotated with CDI an…

### PR DESCRIPTION
**Pull request by tomas-langer**
…nots.

### CODE REVIEW

File: microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java [hunk 1]
```diff
@@ -278,12 +279,40 @@ private SeContainer createContainter(ClassLoader classLoader) {
...
+                    .forEach(initializer::addBeanClasses);
+
             STARTUP_LOGGER.finest("Initializer");
             SeContainer container = initializer.initialize();
             STARTUP_LOGGER.finest("Initalizer.initialize()");
             return container;
         }
 
+        private boolean notACdiBean(Class<?> clazz) {
+            if (clazz.getAnnotation(RequestScoped.class) != null) {
```
Comment (ljnelson) [inline microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java]: `Dependent` too, right?

Also, what about custom scopes?  Those should be rejected as well, right?

File: microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java [hunk 2]
```diff
@@ -278,12 +279,40 @@ private SeContainer createContainter(ClassLoader classLoader) {
                                                               .asOptionalMap()
                                                               .orElse(CollectionsHelper.mapOf()));
             initializer.setProperties(props);
+
+            // add resource classes explicitly configured without CDI annotations
+            this.resourceClasses.stream()
+                    .filter(this::notACdiBean)
+                    .forEach(initializer::addBeanClasses);
```
Comment (ljnelson) [inline microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java]: I think we need to make sure that if we have discovered class `X` but an instance of `X` is reported by `Application#getSingletons()` then probably we shouldn't add `X` here.

We also need to make sure we add the return value of `Application#getSingletons()`, if non-`null` and non-empty, to the CDI container.

File: microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java (hunk 1 above)
Comment (tomas-langer) [reply microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java]: Is there a way to get a list of custom scopes before CDI initialization?

Comment (ljnelson) [reply microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java]: No.  A portable extension, which runs as the container is initializing, may add any it likes.

**Comment by tomas-langer** [comment]
Closing pull request until the approach is clarified with MP.

**Pull request closed without merging**